Runtime/Commands/ActionManager.cs

```diff
@@ -0,0 +1,410 @@
+                var paramMetadata = parameter.GetCustomAttribute<YarnParameterAttribute>();
+                if (paramMetadata != null)
+                {
+                    var methodType = method.DeclaringType;
+                    var injectorMeta = methodType.GetMethod(paramMetadata.Injector, AllStaticMembers);
+                    if (IsInjector(methodType, injectorMeta, targetType))
+                    {
+                        return (Injector)injectorMeta.CreateDelegate(typeof(Injector));
+                    }
+                }
+
+                return arg => GameObject.Find(arg)?.GetComponentInChildren(targetType);
+            }
+
+            // bools can take "true" or "false", or the parameter name.
+            if (typeof(bool).IsAssignableFrom(targetType))
+            {
+                return arg =>
+                {
+                    if (arg.Equals(parameter.Name, StringComparison.InvariantCultureIgnoreCase)) { return true; }
+                    if (bool.TryParse(arg, out bool res)) { return res; }
+                    throw new ArgumentException(
+                        $"Can't convert the given parameter at position {index + 1} (\"{arg}\") to parameter " +
+                        $"{parameter.Name} of type {typeof(bool).FullName}.");
+                };
+            }
+
+            // try converting using IConvertible.
+            return arg =>
+            {
+                try
+                {
+                    return Convert.ChangeType(arg, targetType, CultureInfo.InvariantCulture);
+                }
+                catch (Exception e)
+                {
+                    throw new ArgumentException(
+                        $"Can't convert the given parameter at position {index + 1} (\"{arg}\") to parameter " +
+                        $"{parameter.Name} of type {targetType.FullName}: {e}");
+                }
+            };
+        }
+
+        private static DispatchCommand CreateCommandRunner(
+            MethodInfo method, YarnCommandAttribute metadata, ref Dictionary<string, Injector> injectorCache)
+        {
+            var methodType = method.DeclaringType;
+            if (methodType == null) { throw new ArgumentException($"Method {method.Name} does not have a type...somehow."); }
+
+            Injector injector = null;
+            if (!method.IsStatic)
+            {
+                injector = GetInjectorForMethod(methodType, metadata)
+                    ?? GetInjectorForType(methodType, ref injectorCache)
+                    ?? GetDefaultMonoBehaviourInjector(methodType, metadata.Name);
+            }
+
+            var converters = CreateConverters(method);
+
+            return new DispatchCommand()
+            {
+                Method = method,
+                Injector = injector,
+                Converters = converters
+            };
+        }
+
+        private static Type GetFuncType(int paramCount)
+        {
+            // this causes me physical pain
```

Comment:
I think iOS specifically prohibits codegen (eg MessagePack is forced to AOT because of this). Not sure if dynamically creating a delegate would count?

Reply:
I think Tim means using something like a compile-time code generator, not runtime JIT stuff (which is very much not allowed in AOT environments).

Reply:
Looked into it a little bit, it seems like support isn't 100% there for source generators (because Nuget). I think we should probably leave it for now and revisit it when 2021 goes into LTS.

Reply:
agreed